.github/workflows/CI.yml

```diff
@@ -32,6 +32,4 @@ jobs:
 
     - name: Build and Run Tests
       run: |
-          ./gradlew build
-           # Generated by 'opensearch.pluginzip' custom gradle plugin
```

Comment:
Could we extract separate step for it? 

```
    - name: Generate plugin ZIP distribution
      run: |
           # Generated by 'opensearch.pluginzip' custom gradle plugin
           ./gradlew publishPluginZipPublicationToZipStagingRepository
```

won't change the result but will be more clean I think, wdyt?

Reply:
i don't see any benefit in doing it in two steps. you could also just run `./gradlew publishPluginZipPublicationToZipStagingRepository` alone and it'd include the `build` step as it depends on those artefacts - i just kept `build` there to make it clearer.
i don't expect `publishPluginZipPublicationToZipStagingRepository` to ever fail (at least i don't understand when it could if `build` worked and created the correct files, since it then just assembles them into a `zip` and pushes that to a staging place), so the overhead of the additional step doesn't seem to be very helpful?

Reply:
> i don't expect `publishPluginZipPublicationToZipStagingRepository` to ever fail

The funny part - it was introduced because it was failing. Here is the background:  `PluginZip` plugin adds new publishing targets and was failing Gradle builds (not because of compilation) due to weird tasks input / output dependencies. It may (and will) come up with Gradle updates :(